docs/code_ref/coordinates.rst

```diff
@@ -114,19 +113,72 @@ Both the heliographic frames use latitude, longitude and radius which are access
+
 
 Observer Location Information
 -----------------------------
 
-Both ``Helioprojective`` and ``Heliocentric`` frames are defined by the location of the observer. For example in ``Helioprojective`` the observer is at the origin of the coordinate system. This information is encoded in the ``Helioprojecitve`` and ``Heliocentric`` frames as the attributes `L0`, `B0` and `D0` which are heliographic longitude, latitude and distance from the center of the Sun. These attributes are automatically populated from meta data when coordinate frames are created using map.
-
-It is possible to convert from a ``Helioprojective`` frame with one observer location to another ``Helioprojecitve`` frame with a different observer location, by converting through ``Heliographic``, this does involve making an assumption of the radius of the Sun to calculate the position on the solar sphere. The conversion can be performed as follows::
+Both `~sunpy.coordinates.frames.Helioprojective` and
+`~sunpy.coordinates.frames.Heliocentric` frames are defined by the location of
+the observer. For example in `~sunpy.coordinates.frames.Helioprojective` the
+observer is at the origin of the coordinate system. This information is encoded
+in the `~sunpy.coordinates.frames.Helioprojective` and
+`~sunpy.coordinates.frames.Heliocentric` frames as the ``observer`` attribute,
+which is itself an instance of the
+`~sunpy.coordinates.frames.HeliographicStonyhurst` frame. The default observer
+location is set to the position of the Earth (using
+`~sunpy.coordinates.ephemeris.get_body_heliographic_stonyhurst`) as long as the
+``obstime`` attribute is specified. If the ``obstime`` attribute is not set the
+observer defaults to ``(0°, 0°, 1 AU)`` i.e. the mean position of the Earth. The
+location of the observer is automatically populated from meta data when
+coordinate frames are created using map.
+
+It is possible to convert from a `~sunpy.coordinates.frames.Helioprojective`
+frame with one observer location to another
+`~sunpy.coordinates.frames.Helioprojective` frame with a different observer
+location, by converting through `~sunpy.coordinates.frames.Heliographic`, this
+does involve making an assumption of the radius of the Sun to calculate the
+position on the solar sphere. The conversion can be performed as follows::
 
   # Input coordinate
-  >>> hpc1 = SkyCoord(0*u.arcsec, 0*u.arcsec, frame=frames.Helioprojective)
-  # Define the location of the new observer as a Helioprojective frame
-  >>> hpc_out = sunpy.coordinates.Helioprojective(L0=10*u.deg)
-  # Perform the conversion
+  >>> hpc1 = SkyCoord(0*u.arcsec, 0*u.arcsec, observer="earth", obstime="2017-07-26", frame=frames.Helioprojective)
+  # Define a new Helioprojective frame with a different observer.
+  >>> hpc_out = sunpy.coordinates.Helioprojective(observer="venus", obstime="2017-07-26")
+  # Perform the transformation from one to the other.
   >>> hpc2 = hpc1.transform_to(hpc_out)
```

Comment:
I may be using this wrong... but if I want to know the coordinates of "venus" on carrington longitude, or in Heliographic I would do the following:
```python
>>> hpc_out.observer.transform_to(frames.HeliographicCarrington)
<HeliographicCarrington Coordinate (obstime=2017-07-26 00:00:00): (lon, lat, radius) in (deg, deg, AU)
    ( 1.02845592,  3.17032536,  0.72510629)>
>>> hpc_out.observer.transform_to(frames.HeliographicStonyhurst)
<HeliographicStonyhurst Coordinate (obstime=2017-07-26 00:00:00): (lon, lat, radius) in (deg, deg, AU)
    ( 77.03547231,  3.17032536,  0.72510629)>
```

If we look in [HELIO ILS service](http://helio-vo.eu/services/interfaces/helio-ils_soap8.php?qtype=1&cat1=ils_trajectories&y_from=2017&mo_from=07&d_from=26&h_from=00&mi_from=00&s_from=00&y_to=2017&mo_to=07&d_to=26&h_to=23&mi_to=59&s_to=59&target_obj%5B%5D=Venus&n_show=&format=html&process=1), we get that:


target_obj | time | julian_int | julian_frac | r_hci | long_hci | lat_hci | long_hee | lat_hee | long_carr
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
 VENUS | 2017-07-26 00:00:00.0 | 2457960 | 0.5 | 0.72510974 | -55.983206 | 3.1704421 | 19.771016 | -2.8432442 | 0.988321 
EARTH | 2017-07-26 00:00:00.0 | 2457960 | 0.5 | 1.0156743 | -133.01833 | 5.3170159 | -57.022893 | 0.0018201559 | 283.9532

Radius matches, and so the latitude in `hci` ( I have to find out in the docs what we meant by that, but's the same that our Stonyhurst), but not the longitudes - the Stonyhurst matches when substracting Earth's longitude in `hci` to Venus's. It could be that one of these is based on Aries point γ. Unfortunately most of the HELIO services where I could find this out are not working. Probably from astropy we can get that info too.